pkg/gameservers/migration_test.go

```diff
@@ -0,0 +1,321 @@
+// Copyright 2020 Google LLC All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gameservers
+
+import (
+	"testing"
+	"time"
+
+	agonesv1 "agones.dev/agones/pkg/apis/agones/v1"
+	agtesting "agones.dev/agones/pkg/testing"
+	"github.com/heptiolabs/healthcheck"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/watch"
+	k8stesting "k8s.io/client-go/testing"
+	"k8s.io/client-go/tools/cache"
+)
+
+func TestMigrationControllerIsRunningGameServer(t *testing.T) {
+	fixtures := map[string]struct {
+		setup    func(*corev1.Pod) *corev1.Pod
+		expected bool
+	}{
+		"not a gameserver pod": {
+			setup: func(pod *corev1.Pod) *corev1.Pod {
+				return &corev1.Pod{}
+			},
+			expected: false,
+		},
+		"game server pod": {
+			setup: func(pod *corev1.Pod) *corev1.Pod {
+				return pod
+			},
+			expected: false,
+		},
+		"game server pod that has a nodename": {
+			setup: func(pod *corev1.Pod) *corev1.Pod {
+				pod.Spec.NodeName = "x"
+				return pod
+			},
+			expected: true,
+		},
+		"game server pod that has a node, but is being deleted": {
+			setup: func(pod *corev1.Pod) *corev1.Pod {
+				pod.Spec.NodeName = "z"
+				now := metav1.Now()
+				pod.ObjectMeta.DeletionTimestamp = &now
+				return pod
+			},
+			expected: false,
+		},
+	}
+
+	for k, v := range fixtures {
+		t.Run(k, func(t *testing.T) {
+			gs := &agonesv1.GameServer{ObjectMeta: metav1.ObjectMeta{Name: "test", Namespace: "default"},
+				Spec: newSingleContainerSpec()}
+			gs.ApplyDefaults()
+
+			gsPod, err := gs.Pod()
+			assert.NoError(t, err)
+
+			pod := v.setup(gsPod)
+			assert.Equal(t, v.expected, isActiveGameServerWithNode(pod))
+		})
+	}
+}
+
+func TestMigrationControllerSyncGameServer(t *testing.T) {
+	t.Parallel()
+
+	ipChangeFixture := "99.99.99.99"
+	nodeNameChangeFixture := "nodeChange"
+
+	type expected struct {
+		updated     bool
+		updateTests func(t *testing.T, gs *agonesv1.GameServer)
+		postTests   func(t *testing.T, mocks agtesting.Mocks)
+	}
+	fixtures := map[string]struct {
+		skip     bool
```

Comment:
This doesn't actually seem used. Was it there for development but no longer needed?

Reply:
Exactly. Removing!